website/docs/cli/commands/test.mdx

```diff
@@ -1,12 +1,165 @@
 ---
-description: Part of the ongoing design research for module integration testing.
+description: >-
+    The tofu test command performs integration tests of OpenTofu modules.
 ---
 
 # Command: test
 
-The `tofu test` command is currently serving as part of
-[the module integration testing experiment](/docs/language/modules/testing-experiment).
+The `tofu test` command runs integration tests to ensure intended behavior of your OpenTofu configuration. It creates real
+infrastructure, asserts conditional checks against the attributes of provisioned resources and attempts
+to clean up the testing infrastructure upon completion.
 
-It's not ready for routine use, but if you'd be interested in trying the
-prototype functionality then we'd love to hear your feedback. See the
-experiment details page linked above for more information.
+## Usage
+
+Usage: `tofu test [options]`
+
+The `test` command executes the automated integration tests defined in `.tftest.hcl` files which OpenTofu will read
+from the current configuration and testing directories.
+
+OpenTofu will create real infrastructure according to the definition in the tested configuration. Then, it will
+sequentially run the test blocks to perform conditional checks and assertions against attributes of provisioned resources.
+Upon tests completion, OpenTofu will attempt to clean up created resources.
+
+_Note_: monitor the test output for successful cleanup.
+
+## General Options
+
+* `-test-directory=path` Set the test directory (default: "tests"). OpenTofu will search for test files in the set directory
+and in the directly of the test command execution.
+*Note* that the path shall be relative to the directory where `tofu test` is called.
+
+* `-filter=testfile` Set the test file *path* within the `test-directory`. The option can be used multiple times to specify several files.
+*Note* that the path shall be relative to the directory where `tofu test` is called.
+
+* `-var 'foo=bar'` Set the input variables of the root module. The option can be used multiple times to set several variables.
+
+* `-var-file=filename` Set the path to the file with variables values in addition to the default files terraform.tfvars and *.auto.tfvars.
+The option can be used multiple times to specify several files.
+
+* `-json` Set for test outputs to be formatted as JSON.
+
+* `-no-color` Disable color codes in the command output.
+
+* `-verbose` Print the plan or state for each test run block as it executes.
+
+## Usage examples
+
+### Flat structure
```

Comment:
I feel like the docs now focus too much on the folder structure and about the combinations of `-filter` and `-test-directory`. It's probably enough to explain the interaction between them in a line or two, instead of creating a project structure mainly for that

Reply:
Hey Arel! Thanks for your feedback. The docu is WIP, however I wanted to share enough examples following GWT pattern as a response to this [feedback](https://github.com/opentofu/opentofu/issues/182). IMO if something is not intuitive, it shall be clearly illustrated to facilitate users onboarding. I feel that it's also influenced by the experience level of working with the testing feature. I'm trying to write docu assuming a reader who tries the tool for the first time.